src/procedural-macros.md

```diff
@@ -368,13 +398,15 @@ When passed to a proc-macro
       always represented as their underlying token trees.
 
 r[macro.proc.token.conversion.from-proc_macro]
-When emitted from a proc macro
+When tokens are emitted from a proc macro:
+
 - Punctuation characters are glued into multi-character operators
   when applicable.
 - Single quotes `'` joined with identifiers are glued into lifetimes.
 - Negative literals are converted into two tokens (the `-` and the literal)
   possibly wrapped into a delimited group ([`Group`]) with implicit delimiters
   ([`Delimiter::None`]) when it's necessary for preserving parsing priorities.
+- The `_` identifier is interpreted as the `_` punctuation.
```

Comment:
I don't think the reference should ever call `_` punctuation, it's a (reserved) identifier in all senses except one - `ident` matcher in `macro_rules` doesn't accept it for backward compatibility reasons (but it can likely be changed over an edition).